client/proxy.go

```diff
@@ -0,0 +1,1055 @@
+	}
+
+	return nil
+}
+
+func (cli *VanClient) ProxyRemove(ctx context.Context, proxyName string) error {
+	// TODO: this should return accumulated errors but get throught the whole thing
+
+	if proxyName == "" {
+		return fmt.Errorf("Unable to delete proxy definition, need proxy name")
+	}
+
+	err := cli.proxyStop(ctx, proxyName)
+	if err != nil {
+		return fmt.Errorf("Not able to stop proxy %w", err)
+	}
+
+	err = cli.KubeClient.CoreV1().Secrets(cli.GetNamespace()).Delete(proxyPrefix+proxyName, &metav1.DeleteOptions{})
+	if err != nil {
+		return fmt.Errorf("Unbable to remove proxy secret: %w", err)
+	}
+
+	err = cli.KubeClient.CoreV1().ConfigMaps(cli.GetNamespace()).Delete(proxyPrefix+proxyName, &metav1.DeleteOptions{})
+	if err != nil {
+		return fmt.Errorf("Unbable to remove proxy config map: %w", err)
+	}
+	return nil
+}
+
+func convert(from interface{}, to interface{}) error {
+	data, err := json.Marshal(from)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(data, to)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cli *VanClient) ProxyBind(ctx context.Context, proxyName string, egress types.ProxyBindOptions) error {
+	proxyDir := getDataHome() + "/skupper/" + proxyName
+
+	_, err := getRootObject(cli)
+	if err != nil {
+		return fmt.Errorf("Skupper not initialized in %s", cli.Namespace)
+	}
+
+	configmap, err := kube.GetConfigMap(proxyPrefix+proxyName, cli.GetNamespace(), cli.KubeClient)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve proxy configmap: %w", err)
+	}
+	proxyConfig, err := qdr.GetRouterConfigFromConfigMap(configmap)
+
+	si, err := cli.ServiceInterfaceInspect(ctx, egress.Address)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve service: %w", err)
+	}
+
+	if si == nil {
+		return fmt.Errorf("Unable to proxy bind, service not found for %s", egress.Address)
+	}
+
+	// TODO switch on egress.Protocol
+	endpoint := qdr.TcpEndpoint{
+		Name:    proxyName + "-egress-" + egress.Address,
+		Host:    egress.Host,
+		Port:    egress.Port,
+		Address: egress.Address,
+	}
+	proxyConfig.AddTcpConnector(endpoint)
+
+	// should this be update or write to config map
+	proxyConfig.UpdateConfigMap(configmap)
+	_, err = cli.KubeClient.CoreV1().ConfigMaps(cli.GetNamespace()).Update(configmap)
+	if err != nil {
+		return fmt.Errorf("Failed to update proxy configmap: %w", err)
+	}
+
+	if isActive(proxyName) {
+		mc, err := qdr.MarshalRouterConfig(*proxyConfig)
+		if err != nil {
+			return fmt.Errorf("Failed to marshall router config: %w", err)
+		}
+
+		err = ioutil.WriteFile(proxyDir+"/config/qdrouterd.json", []byte(mc), 0644)
+		if err != nil {
+			return fmt.Errorf("Failed to write config file: %w", err)
+		}
+
+		url, err := ioutil.ReadFile(proxyDir + "/config/url.txt")
+		agent, err := qdr.Connect(string(url), nil)
+		if err != nil {
+			return fmt.Errorf("Agent error adding tcp connector: %w", err)
+		}
+		record := map[string]interface{}{}
```

Comment:
Maybe have `defer agent.Close()` up here, so that connection is not left open if there is an error processing the endpoint?